docs/api/context-bridge.md

```diff
@@ -0,0 +1,138 @@
+
+`Function` values are proxied to the other context and all other values are **copied** and **frozen**.  I.e. Any data / primtives sent in
+the API object become immutable and updates on either side of the bridge do not result in an update on the other side.
+
+**NOTE:** Arrays containing Functions are currently not supported, Functions must not be a child of an array.
+
+An example complex API object is uncluded below.
+
+```javascript
+const { contextBridge } = require('electron')
+
+contextBridge.bindAPIInMainWorld(
+  'electron',
+  {
+    doThing: () => ipcRenderer.send('do-a-thing'),
+    data: {
+      myFlags: ['a', 'b', 'c'],
+      bootTime: 1234
+    },
+    nestedAPI: {
+      evenDeeper: {
+        youCanDoThisAsMuchAsYouWant: {
+          fn: () => ({
+            returnData: 123
+          })
+        }
+      }
+    }
+  }
+)
+```
+
+### API Functions
+
+`Function` values that you bind through the `contextBridge` are proxied through Electron to ensure that contexts remain isolated.  This
+results in some key limitations that we've outlined below.
+
+#### Parameter / Error / Return Type support
+
+Because parameters, errors and return values are **copied** when they are sent over the bridge there are only certain types that can be used.
+At a high level if the type you want to use can be serialized and un-serialized into the same object it will work.  A table of type support
+has been included below for completeness.
+
+| Type | Parameter Support | Return Value Support | Limitations |
+| ---- | ----------------- | -------------------- | ----------- |
+| `String` | ✅ | ✅ | N/A |
+| `Number` | ✅ | ✅ | N/A |
+| `Boolean` | ✅ | ✅ | N/A |
+| `Object` | ✅ | ✅ | Keys and values must be supported in this table.  Prototype modifications are dropped.  Sending custom classes will copy values but not the prototype. |
+| `Error` | ✅ | ✅ | Errors that are thrown are also copied, this can result in the message and stack trace of the error changing slightly due to being thrown in a different context |
+| `Promise` | ✅ | ✅ | Promises are proxied through the bridge **only** if it is the return value, promises as function parameters are dropped |
+| `Function` | ❌ | ❌ | N/A |
+| `Symbol` | ❌ | ❌ | Symbols cannot be copied across contexts so they are dropped |
+
+
+If the type you care about is not in the above table it is probably not supported.
+
+### Reverse Bindings
+
+The `contextBridge.bindAPIInMainWorld` allows you to expose APIs from the **Isolated World** the preload script runs in to the **Main World** the renderer runs in.  If you want
+to allow the main world to provide an API back to the isolated world you can set the `allowReverseBinding` option to `true`.  By setting that option the API exposed in the main
+world will have a new method injected on it on the `bind` key.  You can use this method by providing it a single `api` object in the same structure as the original `bindAPIInMainWorld`
```

Comment:
If `bind` already exists, I assume we'll just override it.  We may want to error / warn if that happens.

> You can use this method by providing it a single `api` object in the same structure as the original `bindAPIInMainWorld`

I'd use "same limitations" rather than "same structure" since structure could also be interpreted as "the same keys and values"